resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsLifecycleManager.scala

```diff
@@ -217,14 +217,42 @@ private[spark] class ExecutorPodsLifecycleManager(
     ExecutorExited(exitCode, exitCausedByApp, exitMessage)
   }
 
+  // A utility function to try and help people figure out whats gone wrong faster.
+  private def describeExitCode(code: Int): String = {
```

Comment:
What about moving this into `object ExecutorPodsLifecycleManager` and making it accessible for the `ExecutorPodsLifecycleManagerSuite` to avoid the copy-paste?